pkg/workloadmeta/store.go

```diff
@@ -402,20 +402,12 @@ func (s *store) handleEvents(evs []CollectorEvent) {
 		}
 	}
 
-	// unlock the store before notifying subscribers, as they might need to
-	// read it for related entities (such as a pod's containers) while they
-	// process an event.
-	s.storeMut.Unlock()
+	filteredEvents := make(map[subscriber][]Event)
```

Comment:
Nit: since we know how many subscribers we have in advance
```suggestion
	filteredEvents := make(map[subscriber][]Event, len(s.subscribers))
```